File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.py
class Solution(object):
    def addSpaces(self, s, spaces):
        """
        :type s: str
        :type spaces: List[int]
        :rtype: str
        """

        size = len(s)
        it = 0
        ans = []

        for i in range(size):
            if it < len(spaces) and i == spaces[it]:
                it += 1
                ans.append(" ")
            ans.append(s[i])

        return "".join(ans)
        

File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        int size = s.length();
        int it = 0;
        string ans = "";

        for(int i = 0; i < size; i++){

            if(it < spaces.size()){
                if(i == spaces[it]){
                    it++;
                    ans += " ";
                }
            }

            ans += s[i];
        }

        return ans;
    }
};